### PR TITLE
impl pulp variant

### PR DESCRIPTION
**Pull request by sarah-quinones**
timings on my desktop (single threaded, `11th Gen Intel(R) Core(TM) i5-11400 @ 2.60GHz`)
```
+ hyperfine --warmup 1 'target/release/mandelbrot 3200 3200 --algo scalar'
Benchmark 1: target/release/mandelbrot 3200 3200 --algo scalar
  Time (mean ± σ):     774.5 ms ±  42.6 ms    [User: 760.9 ms, System: 13.0 ms]
  Range (min … max):   724.2 ms … 868.9 ms    10 runs

+ hyperfine --warmup 1 'target/release/mandelbrot 3200 3200 --algo simd'
Benchmark 1: target/release/mandelbrot 3200 3200 --algo simd
  Time (mean ± σ):     142.5 ms ±   3.3 ms    [User: 135.8 ms, System: 6.5 ms]
  Range (min … max):   138.3 ms … 149.4 ms    20 runs

+ hyperfine --warmup 1 'target/release/mandelbrot 3200 3200 --algo autovect'
Benchmark 1: target/release/mandelbrot 3200 3200 --algo autovect
  Time (mean ± σ):     390.6 ms ±  11.0 ms    [User: 384.5 ms, System: 6.1 ms]
  Range (min … max):   376.7 ms … 415.6 ms    10 runs

+ hyperfine --warmup 1 'target/release/mandelbrot 3200 3200 --algo wide'
Benchmark 1: target/release/mandelbrot 3200 3200 --algo wide
  Time (mean ± σ):     266.6 ms ±   2.6 ms    [User: 259.0 ms, System: 7.2 ms]
  Range (min … max):   262.3 ms … 270.5 ms    11 runs

+ hyperfine --warmup 1 'target/release/mandelbrot 3200 3200 --algo pulp'
Benchmark 1: target/release/mandelbrot 3200 3200 --algo pulp
  Time (mean ± σ):     142.4 ms ±   3.3 ms    [User: 135.9 ms, System: 6.5 ms]
  Range (min … max):   138.3 ms … 150.7 ms    20 runs
```

### CODE REVIEW

**Comment by itamarst** [comment]
Thank you, numbers look great!

File: Cargo.toml [hunk 1]
```diff
@@ -10,6 +10,8 @@ rayon = "^1.0"
 ispc = { version = "^1.0.4", optional = true }
 structopt = { version = "0.3.0", features = ["color"] }
 wide = "0.7.28"
+pulp = { version = "0.20.1", features = ["nightly"] }
```
Comment (itamarst) [inline Cargo.toml:13]: Does `pulp` require nightly? Part of my motivation here is to showcase libraries that _don't_ require nightly, i.e. where `std::simd` isn't an option.

Comment (sarah-quinones) [reply Cargo.toml:13]: it doesn't. it only requires it for avx512 support (until it's stabilized)

i wanted it so the benchmark would be a fair fight

Comment (sarah-quinones) [reply Cargo.toml:13]: the code should still work as-is without the nightly flag, it'll just be slower on avx512 machines

Comment (itamarst) [reply Cargo.toml:13]: Makes sense, thanks for the explanation.

**Pull request merged**